packages/cactus-plugin-ledger-connector-iroha2/src/main/typescript/plugin-ledger-connector-iroha2.ts

```diff
@@ -0,0 +1,629 @@
+
+    // Merge default config with config passed to this function
+    const mergedConfig = { ...this.defaultConfig, ...baseConfig };
+
+    if (!mergedConfig.torii) {
+      throw new Error("torii is missing in combined configuration");
+    }
+
+    // Parse signing key pair
+    let keyPair: KeyPair | undefined;
+    if (mergedConfig.signingCredential) {
+      keyPair = await this.getSigningKeyPair(mergedConfig.signingCredential);
+    }
+
+    // Parse account ID
+    let accountId: AccountId | undefined;
+    if (mergedConfig.accountId) {
+      accountId = AccountId({
+        name: mergedConfig.accountId.name,
+        domain_id: DomainId({
+          name: mergedConfig.accountId.domainId,
+        }),
+      });
+    }
+
+    // TODO - confirm which args are optional and remove type casts accordingly
+    return new CactusIrohaV2Client(
+      {
+        apiURL: mergedConfig.torii.apiURL as string,
+        telemetryURL: mergedConfig.torii.telemetryURL as string,
+      },
+      [accountId as AccountId, keyPair as KeyPair],
+      this.options.logLevel,
+    );
+  }
+
+  /**
+   * Helper function used to safely check that required number of parameters were supplied in the request.
+   *
+   * @param params Parameter list from the request.
+   * @param expectedCount Expected parameter count
+   * @param functionName Function that needs specified number of args (used for error logging only)
+   *
+   * @returns List of checked parameters (of `expectedCount` length).
+   */
+  private checkArgsCount(
+    params: unknown[] | undefined,
+    expectedCount: number,
+    functionName: string,
+  ): unknown[] {
+    if (!params) {
+      throw new Error(
+        `Error [${functionName}] - Missing required parameters in request.`,
+      );
+    }
+
+    const requiredParams = params.slice(0, expectedCount);
+
+    if (requiredParams.length < expectedCount) {
+      throw new Error(
+        `Error [${functionName}] - No enough parameters. Expected: ${expectedCount}, got: ${requiredParams.length}`,
+      );
+    }
+
+    return requiredParams;
+  }
+
+  /**
+   * Validate required parameters and call transact method
+   * (will add instruction to the list of operations to be executed)
+   *
+   * @note `expectedCount` must be equal to number of args required by `transactFunction`,
+   * otherwise the code will not compile (this is intended safety-check)
+   *
+   * @param client `CactusIrohaV2Client` object.
+   * @param transactFunction Transact function to be executed
+   * @param params Parameter list from the request.
+   * @param expectedCount Expected parameter count
+   */
+  private addTransactionWithCheckedParams<
+    T extends (...args: any[]) => unknown
+  >(
+    client: CactusIrohaV2Client,
+    transactFunction: T,
+    params: unknown[] | undefined,
+    expectedCount: LengthOf<Parameters<T>>,
+  ): void {
+    transactFunction.apply(
+      client,
+      this.checkArgsCount(params, expectedCount, transactFunction.name),
```

Comment:
@outSH Please try to avoid nesting function calls inline where possible. It's easier to make the code read like English prose if you store the output of the checkArgsCount call in a variable that hints at what it is in there.

Reply:
@petermetz OK, the response of `checkArgsCount` can't be infered from the name so separate var will help.

Reply:
@outSH Thank you!